subsys/zbus/zbus.c

```diff
@@ -380,10 +380,8 @@ int zbus_chan_pub(const struct zbus_channel *chan, const void *msg, k_timeout_t
 
 	_ZBUS_ASSERT(chan != NULL, "chan is required");
 	_ZBUS_ASSERT(msg != NULL, "msg is required");
-
-	if (k_is_in_isr()) {
-		timeout = K_NO_WAIT;
-	}
+	_ZBUS_ASSERT(!(k_is_in_isr() && !K_TIMEOUT_EQ(timeout, K_NO_WAIT)),
+		     "Timeout must be K_NO_WAIT inside an ISR");
```

Comment:
Just to keep consistency with the other asserts, please change the text. 
```suggestion
		     "timeout must be K_NO_WAIT inside an ISR");
```